monitoring/tests/test_collectors.py

```diff
@@ -0,0 +1,238 @@
+#!/usr/bin/env python3
+import unittest
+import requests_mock
+from datetime import datetime, timezone
+from collectors.jenkins_collector_adapter import JenkinsCollectorAdapter, InvalidResponseCodeError
+from stats import BuildStats, StageStats
+
+
+class TestJenkinsGetBuildStatsEndpoint(unittest.TestCase):
+    def test_build_stats_endpoint_is_good(self):
+        url = JenkinsCollectorAdapter.get_build_stats_endpoint(build_url='http://localhost:8080/job/MyJob/1')
+        self.assertEqual(url, 'http://localhost:8080/job/MyJob/1/wfapi/describe')
+
+
+class TestJenkinsGetRawStats(unittest.TestCase):
+    def test_get_raw_stats_ok(self):
+        with requests_mock.mock() as m:
+            m.get('http://localhost:8080/job/MyJob/1/wfapi/describe', json={
+                'id': 1,
+                'status': 'SUCCESS',
+                'durationMillis': 1000,
+                'endTimeMillis': 1234,
+            })
+
+            collector = JenkinsCollectorAdapter('MyJob', 'http://localhost:8080/job/MyJob/1')
+            stats = collector.get_raw_stats_from_jenkins()
+
+        self.assertIsNotNone(stats)
+        self.assertEqual(stats['id'], 1)
+        self.assertEqual(stats['endTimeMillis'], 1234)
+        self.assertEqual(stats['status'], 'SUCCESS')
+        self.assertEqual(stats['durationMillis'], 1000)
+
+    def test_get_raw_stats_error(self):
```

Comment:
Pedantic, but we should probably name it `test_get_raw_stats_invalid_build_id` or something.